diff -ru old-pycaml/pycaml_stubs.c chemoelectric-pycaml-8614105/pycaml_stubs.c
--- old-pycaml/pycaml_stubs.c	2012-08-15 23:03:45.604693537 +0200
+++ chemoelectric-pycaml-8614105/pycaml_stubs.c	2012-08-16 22:03:01.969958132 +0200
@@ -1886,7 +1886,6 @@
     CAMLparam1(array);
     PyObject *tuple = PyTuple_New(Wosize_val(array));
     int i;
-    int x;
 
     for( i = 0; i < Wosize_val(array); i++ )
       {
@@ -1904,7 +1903,7 @@
 	   does this of its own! Nice, isn't it?
 	*/
 	Py_INCREF(entry);
-	x = PyTuple_SetItem(tuple,i,entry);
+	PyTuple_SetItem(tuple,i,entry);
       }
 
     CAMLreturn(pywrap_steal(tuple));
@@ -1973,7 +1972,6 @@
     CAMLparam1(array);
     PyObject *list = PyList_New(Wosize_val(array));
     int i;
-    int x;
 
     for( i = 0; i < Wosize_val(array); i++ )
       {
@@ -1981,7 +1979,7 @@
 	entry=pyunwrap(Field(array,i));
 	/* T.F.: See pytuple_fromarray code comments! */
 	Py_INCREF(entry);
-	x = PyList_SetItem(list,i,entry);
+	PyList_SetItem(list,i,entry);
       }
     CAMLreturn(pywrap_steal(list));
 }
