diff -ru ./chemoelectric-pycaml-8614105/pycaml.ml ./pyunicodeapichanges/pycaml.ml
--- chemoelectric-pycaml-8614105/pycaml.ml	2012-06-19 11:37:41.480033019 +0200
+++ pyunicodeapichanges/pycaml.ml	2012-06-19 11:34:42.464040503 +0200
@@ -434,15 +434,6 @@
 (* TypeUTF8Decoder *)
 external pyunicode_decodeutf8 : (string * string option) -> pyobject = "PyUnicode_DecodeUTF8_wrapper"
 
-(* TypeUTF16Decoder *)
-external pyunicode_decodeutf16 : (string * string option * int option) -> pyobject = "PyUnicode_DecodeUTF16_wrapper"
-external pyunicode_decodeutf32 : (string * string option * int option) -> pyobject = "PyUnicode_DecodeUTF32_wrapper"
-
-(*-----------------------------------------------------------------------*)
-
-external pyunicode_fromunicode : (int -> int) -> int -> pyobject = "PyUnicode_FromUnicode_wrapper"
-external pyunicode_asunicode   : pyobject -> int array = "PyUnicode_AsUnicode_wrapper"
-
 (*-----------------------------------------------------------------------*)
 
 external pytuple_fromarray : pyobject array -> pyobject = "pytuple_fromarray"
diff -ru ./chemoelectric-pycaml-8614105/pycaml_stubs.c ./pyunicodeapichanges/pycaml_stubs.c
--- chemoelectric-pycaml-8614105/pycaml_stubs.c	2012-06-19 11:37:41.481033018 +0200
+++ pyunicodeapichanges/pycaml_stubs.c	2012-06-19 11:34:42.463040503 +0200
@@ -1532,7 +1532,7 @@
 TypeUTF8Decoder(PyUnicode_DecodeUTF8, pywrap_steal)
 
 /*-----------------------------------------------------------------------*/
-
+#if 0
 #define TypeUTF16Decoder(func, wrap_obj)                                \
     CAMLprim value func##_wrapper(value py_args)                        \
     {                                                                   \
@@ -1606,7 +1606,7 @@
 
     CAMLreturn(result);
 }
-
+#endif
 /*-----------------------------------------------------------------------*/
 
 /* Value -> Pyobject */
